classes/stock/StockAvailable.php

```diff
@@ -107,8 +107,29 @@ public function updateWs()
         if ($this->depends_on_stock) {
             return WebserviceRequest::getInstance()->setError(500, $this->trans('You cannot update the available stock when it depends on stock.', [], 'Admin.Catalog.Notification'), 133);
         }
+        $result = $this->update();
+        if ($this->id_product_attribute == 0) {
+            $id_shop = (Shop::getContext() != Shop::CONTEXT_GROUP && $this->id_shop ? $this->id_shop : null);
+            $quantity = (int) Db::getInstance(_PS_USE_SQL_SLAVE_)->getValue(
+                'SELECT quantity as quantity ' .
+                'FROM ' . _DB_PREFIX_ . 'stock_available ' .
+                'WHERE id_product = ' . (int) $this->id_product . ' ' .
+                'AND id_product_attribute = 0 ' .
+                StockAvailable::addSqlShopRestriction(null, $id_shop)
+            );
+        
+            Hook::exec(
```

Comment:
This hook must be outside the condition, using `$this->product_attribute` and allow to change `$result`

Reply:
Please look how the methods are called one from another: `updateWS()` ->`update()` -> `postSave()` -> `seQuantity()` .
Normally this hook is called in `setQuantity()`, but for `id_product_attribute == 0` it is not called because of the condition
`if ($this->id_product_attribute == 0) {
            return true;
        }`
in `postSave()`
The idea standing behind the change proposed by me is to call this hook in `updateWS()` only for `id_product_attribute == 0` because for other cases the hook is called in `setQuantity()`